superset/sql/parse.py

```diff
@@ -367,6 +367,27 @@ def get_settings(self) -> dict[str, str | bool]:
             for eq in set_item.find_all(exp.EQ)
         }
 
+    def get_predicates(self) -> list[exp.Predicate]:
```

Comment:
Can you add this to the base class as well? A dummy implementation raising `NotImplementedError`.